lib/destroyed_at.rb

```diff
@@ -39,12 +39,25 @@ def destroyed(time = nil)
 
   # Set an object's destroyed_at time.
   def destroy(timestamp = nil)
-    timestamp ||= @marked_for_destruction_at || current_time_from_proper_timezone
-    raw_write_attribute(:destroyed_at, timestamp)
-    run_callbacks(:destroy) do
-      destroy_associations
-      self.class.unscoped.where(self.class.primary_key => id).update_all(destroyed_at: timestamp)
-      @destroyed = true
+    transaction do
+      timestamp ||= @marked_for_destruction_at || current_time_from_proper_timezone
+      raw_write_attribute(:destroyed_at, timestamp)
+
+      run_callbacks(:destroy) do
+        destroy_associations
+        self.class.unscoped.where(self.class.primary_key => id).update_all(destroyed_at: timestamp)
+        @destroyed = true
+
+        next unless ActiveRecord::VERSION::STRING >= '4.2'
```

Comment:
👋 @michaeldupuisjr. You're right—they're actually equivalent though using `next` here replaces those conditionals with [*guard clauses*](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals).

`next` is also preferred here over `return` since there's a callback chain where the `after_` callback will be executed.

Sidenote: I added another commit to return `@destroyed`.